backend/asmgen.ml

```diff
@@ -170,21 +170,63 @@ let ocamlcfg_verbose =
   | Some "1" -> true
   | Some _ | None -> false
 
+let recompute_liveness_on_cfg (cfg_with_layout : Cfg_with_layout.t) : Cfg_with_layout.t =
+  let cfg = Cfg_with_layout.cfg cfg_with_layout in
+  Cfg.iter_blocks cfg ~f:(fun _label block ->
+      let canary = Reg.create Cmm.Val in
+      canary.Reg.raw_name <- Reg.Raw_name.create_from_var (Ident.create_local "canary");
+      let canary_singleton = Reg.Set.singleton canary in
+      ListLabels.iter block.body ~f:(fun instr -> instr.Cfg.live <- canary_singleton);
+      block.terminator.Cfg.live <- canary_singleton);
+  begin match Cfg_liveness.Liveness.run cfg ~init:Reg.Set.empty () with
+    | Result.Ok (_ : Cfg_liveness.Liveness.map) -> ()
+    | Result.Error (_ : Cfg_liveness.Liveness.map) ->
+      Misc.fatal_errorf "Unable to compute liveness from CFG for function %s@."
+        cfg.Cfg.fun_name;
+  end;
+  Cfg.iter_blocks cfg ~f:(fun _label block ->
+      block.body <- ListLabels.filter block.body ~f:(fun instr ->
+          not (Cfg.is_noop_move instr)));
+  let layout : Label.t list =
```

Comment:
It is not the liveness computation per se, it would be the
block merge and/or reachability analysis that comes after.
I don't have an example at hand, though.